Private is missing from ProjectReference

When I parse a project, some properties are missing in the references of the project.

### What You Are Seeing?
In the ProjectReference object, I have theses properties:
* FilePath FilePath
* string Name
* FilePath Package
* string Project

### What is Expected?
I should have also the property Private (also present in ProjectAssemblyReference)

### What version of Cake are you using?
 Version 0.19.3+Branch.main.Sha.c8a44d4c031e5c9631ec3044536c6af70f3fb04f

### Are you running on a 32 or 64 bit system?
64 bit

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)
Sets Copy local to false or true to a project of your solution.
Then create a build.cake file with:
```CSharp
var parsedProject = ParseProject("<your project path here>");
foreach(var prjRef in parsedProject.ProjectReferences){
    prjRef.Dump();
}
```

@augustoproiete 

I am not sure how to recreate this scenario.
```
Sets Copy local to false or true to a project of your solution.
```
How to set this? I couldn't find this copy local variable on project/solution.

```
var parsedProject = ParseProject("<your project path here>");
foreach(var prjRef in parsedProject.ProjectReferences){
    prjRef.Dump();
}
```
Just with above code in cake file I am getting a different error.

```
Tool 'cake.tool' (version '1.1.0') was restored. Available commands: dotnet-cake

Restore was successful.
C:/CodeContribute/exampleapp/build.cake(3,12): error CS1061: 'ProjectReference' does not contain a definition for 'Dump' and no accessible extension method 'Dump' accepting a first argument of type 'ProjectReference' could be found (are you missing a using directive or an assembly reference?)
Error: Error(s) occurred when compiling build script:
C:/CodeContribute/exampleapp/build.cake(3,12): error CS1061: 'ProjectReference' does not contain a definition for 'Dump' and no accessible extension method 'Dump' accepting a first argument of type 'ProjectReference' could be found (are you missing a using directive or an assembly reference?)
```

--- a/src/Cake.Common.Tests/Properties/Resources.resx
+++ b/src/Cake.Common.Tests/Properties/Resources.resx
@@ -798,6 +798,11 @@ Line #3]]&gt;&lt;/releaseNotes&gt;
       &lt;Project&gt;{ABC3F1CB-F84E-43ED-A120-0CCFE344D250}&lt;/Project&gt;
       &lt;Name&gt;Cake.Common&lt;/Name&gt;
     &lt;/ProjectReference&gt;
+    &lt;ProjectReference Include="..\Cake.Common\Cake.Core.csproj"&gt;
+      &lt;Project&gt;{6C54B9D8-A47D-45D8-82EF-3ADAF7FDD530}&lt;/Project&gt;
+      &lt;Name&gt;Cake.Core&lt;/Name&gt;
+      &lt;Private&gt;True&lt;/Private&gt;
+    &lt;/ProjectReference&gt;
   &lt;/ItemGroup&gt;
   &lt;Import Project="$(MSBuildExtensionsPath32)\Microsoft\Portable\$(TargetFrameworkVersion)\Microsoft.Portable.CSharp.targets" /&gt;
 &lt;/Project&gt;</value>

--- a/src/Cake.Common.Tests/Unit/Solution/Project/ProjectParserTests.cs
+++ b/src/Cake.Common.Tests/Unit/Solution/Project/ProjectParserTests.cs
@@ -235,11 +235,13 @@ namespace Cake.Common.Tests.Unit.Solution.Project
                 var result = fixture.Parse();
 
                 // Then
-                Assert.Equal(1, result.ProjectReferences.Count);
+                Assert.Equal(2, result.ProjectReferences.Count);
                 Assert.Equal("/Working/../Cake.Common/Cake.Common.csproj", result.ProjectReferences.First().FilePath.FullPath);
                 Assert.Equal("..\\Cake.Common\\Cake.Common.csproj", result.ProjectReferences.First().RelativePath);
                 Assert.Equal("{ABC3F1CB-F84E-43ED-A120-0CCFE344D250}", result.ProjectReferences.First().Project);
                 Assert.Equal("Cake.Common", result.ProjectReferences.First().Name);
+                Assert.Equal(null, result.ProjectReferences.First().Private);
+                Assert.Equal(true, result.ProjectReferences.ElementAt(1).Private);
             }
         }
     }

--- a/src/Cake.Common/Solution/Project/ProjectParser.cs
+++ b/src/Cake.Common/Solution/Project/ProjectParser.cs
@@ -183,6 +183,7 @@ namespace Cake.Common.Solution.Project
                  let nameElement = element.Element(ProjectXElement.Name)
                  let projectElement = element.Element(ProjectXElement.Project)
                  let packageElement = element.Element(ProjectXElement.Package)
+                 let privateElement = element.Element(ProjectXElement.Private)
                  select new ProjectReference
                  {
                      FilePath = filePath,
@@ -190,7 +191,8 @@ namespace Cake.Common.Solution.Project
                      Name = nameElement?.Value,
                      Project = projectElement?.Value,
                      Package = string.IsNullOrEmpty(packageElement?.Value)
-                        ? null : rootPath.CombineWithFilePath(packageElement.Value)
+                        ? null : rootPath.CombineWithFilePath(packageElement.Value),
+                     Private = privateElement == null ? (bool?)null : bool.Parse(privateElement.Value),
                  }).ToArray();
 
             return new ProjectParserResult(

--- a/src/Cake.Common/Solution/Project/ProjectReference.cs
+++ b/src/Cake.Common/Solution/Project/ProjectReference.cs
@@ -54,5 +54,13 @@ namespace Cake.Common.Solution.Project
         /// The path of the project file that is being referenced.
         /// </value>
         public FilePath Package { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the reference should be copied to the output folder.
+        /// </summary>
+        /// <value>
+        /// Whether the reference should be copied to the output folder.
+        /// </value>
+        public bool? Private { get; set; }
     }
 }

